### PR TITLE
AT: Add theme upload size check

### PR DESCRIPTION
**Pull request by lamosty**
Currently, we only check for the uploaded theme file size on the server but do not respond back with an error if the uploaded theme has bigger size. In this PR, we add a client-side check for the uploaded theme size so users know more info.

We also make the failure message for failing to upload a theme better.

## Testing instructions

1. Navigate to http://calypso.localhost:3000/themes/upload/<site> and try to upload a file which is over 50MB. You should get an error saying that the theme zip is too big.
2. Try uploading a proper theme to verify that AT works

### CODE REVIEW

File: client/my-sites/themes/theme-upload/index.jsx [hunk 1]
```diff
@@ -161,6 +167,14 @@ class Upload extends React.Component {
 		const file = files[ 0 ] || files.item( 0 );
 		debug( 'zip file:', file );
 
+		if ( file.size && file.size > MAX_UPLOADED_THEME_SIZE ) {
```
Comment (jhnstn) [inline client/my-sites/themes/theme-upload/index.jsx]: Do you need the property existence check here? `( undefined > MAX_UPLOADED_THEME_SIZE ) === false`

Comment (lamosty) [inline client/my-sites/themes/theme-upload/index.jsx]: Probably not but I always do this, especially in PHP. Maybe I've spent too much time in PHP now.. :D

File: client/my-sites/themes/theme-upload/index.jsx [hunk 2]
```diff
@@ -161,6 +167,14 @@ class Upload extends React.Component {
 		const file = files[ 0 ] || files.item( 0 );
 		debug( 'zip file:', file );
 
+		if ( file.size > MAX_UPLOADED_THEME_SIZE ) {
+			notices.error(
+				translate( 'Theme zip is too big. Max theme size is 50 MB.' )
```
Comment (gwwar) [inline client/my-sites/themes/theme-upload/index.jsx]: Maybe `Theme zip is too large. Please upload a theme under 50 MB.`

Also it appears that there's a too large string above that says it should be 10MB. Is that a different case or should it be updated to 50MB?

Comment (lamosty) [inline client/my-sites/themes/theme-upload/index.jsx]: I haven't read the whole code but I suppose that's for Jetpack sites or after the site was transferred (and thus made a JP site). There is only one error if the AT initiate endpoint fails and it is `initiate_failure`: https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/actions.js#L657-L657.

Comment (lamosty) [inline client/my-sites/themes/theme-upload/index.jsx]: And thanks for the updated message.

**Pull request merged**